File: javascript/1351-count-negative-numbers-in-a-sorted-matrix.js
/**
 * @param {number[][]} grid
 * @return {number}
 */

const countNegatives = function(grid) {
    const gridFlat = grid.flat()
    let count = 0
    gridFlat.forEach(num => { if (num < 0) count++ })
    return count
};

const grid = [
    [4, 3, 2, -1],
    [3, 2, 1, -1],
    [1, 1, -1, -2],
    [-1, -1, -2, -3],
];

console.log(countNegatives(grid))

File: python/1351-count-negative-numbers-in-a-sorted-matrix.py

from typing import List


class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        grid_flat = [ele for col in grid for ele in col]
        count = len([ele for ele in grid_flat if ele < 0])
        return count
               
               
grid: List[List[int]] = [
    [4, 3, 2, -1],
    [3, 2, 1, -1],
    [1, 1, -1, -2],
    [-1, -1, -2, -3],
]
print(Solution().countNegatives(grid))
